src/input-helper.ts

```diff
@@ -110,7 +106,7 @@ export function getInputs(): IGitSourceSettings {
   core.debug(`recursive submodules = ${result.nestedSubmodules}`)
 
   // Auth token
-  result.authToken = core.getInput('token')
+  result.authToken = core.getInput('token', {required: true})
```

Comment:
Isn't this a breaking change? Does the action need a token if checking out a public repo?

If we're making this required, we should update the action.yml file with `required: true` for this input as well.

Reply:
Yes today it already needs a token but fails with a bad error #221

Reply:
Generally it doesnt need to be specified, because the default value works but sometimes folks have set it to `''` or due to fork PR a secret doesnt get mapped in

Reply:
If it's empty, can I still checkout from a public repo?

Reply:
No today it fails like this:

```
/usr/bin/git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin +559278e5d4074a5495dceca4d32edaa1acc01724:refs/remotes/origin/master
##[error]fatal: could not read Username for 'https://github.com': terminal prompts disabled
```